xcm/xcm-builder/src/location_conversion.rs

```diff
@@ -147,6 +147,39 @@ impl<Network: Get<Option<NetworkId>>, AccountId: From<[u8; 20]> + Into<[u8; 20]>
 	}
 }
 
+/// Tries to convert **foreign** global consensus parachain to accountId.
+///
+/// **foreign** means `parents > 1`
+///
+/// (E.g.: can be used for sovereign account conversion)
+pub struct GlobalConsensusParachainConvert<AccountId>(PhantomData<AccountId>);
+impl<AccountId: From<[u8; 32]> + Clone> Convert<MultiLocation, AccountId>
+	for GlobalConsensusParachainConvert<AccountId>
+{
+	fn convert_ref(location: impl Borrow<MultiLocation>) -> Result<AccountId, ()> {
+		match location.borrow() {
+			MultiLocation {
+				parents,
+				interior: X2(GlobalConsensus(network), Parachain(para_id)),
+			} if parents > &1_u8 =>
+				Ok(AccountId::from(GlobalConsensusParachainConvert::<AccountId>::from_params(
+					network, para_id, *parents,
+				))),
+			_ => Err(()),
+		}
+	}
+
+	fn reverse_ref(_: impl Borrow<AccountId>) -> Result<MultiLocation, ()> {
+		// if this will be needed, we could implement some kind of guessing, if we have configuration for supported foreign networkId+paraId
+		Err(())
+	}
+}
+impl<AccountId> GlobalConsensusParachainConvert<AccountId> {
+	fn from_params(network: &NetworkId, para_id: &u32, parents: u8) -> [u8; 32] {
+		(network, para_id, parents).using_encoded(blake2_256)
```

Comment:
Including the parent here looks weird to me, especially when we know that the interior of the `MultiLocation` has to start with a `GlobalConsensus`, because we know that there's no way that the number of parents can change.

Reply:
ok, thank you, make sense, that doesnt matter if you see `(networkId+paraId)` from 1,2,3... parents, still we assume that it is just the only one `(networkId+paraId)`